### PR TITLE
[SPARK-8570] [MLlib] [Docs] Improve MLlib Local Matrix Documentation.

### PR DESCRIPTION
**Pull request by dusenberrymw**
Updated MLlib Data Types Local Matrix section to include information on sparse matrices, added sparse matrix examples to the Scala and Java examples, and added Python examples for both dense and sparse matrices.

### CODE REVIEW

File: docs/mllib-data-types.md [hunk 1]
```diff
@@ -226,7 +226,8 @@ examples = MLUtils.loadLibSVMFile(sc, "data/mllib/sample_libsvm_data.txt")
 
 A local matrix has integer-typed row and column indices and double-typed values, stored on a single
 machine.  MLlib supports dense matrices, whose entry values are stored in a single double array in
-column major.  For example, the following matrix `\[ \begin{pmatrix}
+column-major order, and sparse matrices, whose non-zero entry values are stored in the Compressed Sparse
+Column (CSC) format in column-major order.  For example, the following dense matrix `\[ \begin{pmatrix}
```
Comment (srowen) [inline docs/mllib-data-types.md]: I don't really disagree with documenting this, but is it material to the caller? to commit to that representation? I think you can argue it matters, just checking.

Comment (dusenberrymw) [inline docs/mllib-data-types.md]: Well I think documenting that we are using CSC representation gives the user some reasoning for how the `colPtrs`, `rowIndices`, and `values` parameters interact, and how to use them properly.  Now whether we should *require* that only those parameters be used, as opposed to `values` & `(row, col)` tuples, for example, is another question, and may be worth exploring.

**Comment by dusenberrymw** [comment]
Jenkins, retest this please.

**Comment by dusenberrymw** [comment]
@srowen Any thoughts on this?  I've run it twice now, and it finishes in about 10 minutes (as opposed to the normal ~2.5 hours), and doesn't seem to be running any unit tests.  Here's the end of the output:

```
========================================================================
Running Spark unit tests
========================================================================
Archiving unit tests logs...
> No log files found.
Attempting to post to Github...
 > Post successful.
Archiving artifacts
WARN: No artifacts found that match the file pattern "**/target/unit-tests.log". Configuration error?
WARN: java.lang.InterruptedException: no matches found within 10000
Recording test results
ERROR: Publisher 'Publish JUnit test result report' failed: No test report files were found. Configuration error?
Finished: FAILURE
```

**Comment by dusenberrymw** [comment]
Jenkins, retest this please.

**Comment by sujkh85** [comment]
NAVER - http://www.naver.com/
--------------------------------------------

sujkh@naver.com 님께 보내신 메일 <Re: [spark] [SPARK-8570] [MLlib] [Docs] Improve MLlib Local Matrix Documentation. (#6958)> 이 다음과 같은 이유로 전송 실패했습니다.

--------------------------------------------

받는 사람이 회원님의 메일을 수신차단 하였습니다. 


--------------------------------------------

**Comment by dusenberrymw** [comment]
@srowen Any thoughts on this?

**Comment by dusenberrymw** [comment]
Jenkins, retest this please.

**Comment by dusenberrymw** [comment]
cc @mengxr @jkbradley

**Comment by mengxr** [comment]
@dusenberrymw I will take a look tomorrow.

**Comment by srowen** [comment]
LGTM. It doesn't run tests since it's a doc-only change.

File: docs/mllib-data-types.md (hunk 1 above)
Comment (mengxr) [inline docs/mllib-data-types.md]: minor: We could provide a wiki link for `CSC`: https://en.wikipedia.org/wiki/Sparse_matrix#Compressed_sparse_column_.28CSC_or_CCS.29

**Comment by mengxr** [comment]
LGTM2. Merged into master. Thanks! I made a minor comment on `CSC`. We could add it in a follow-up PR, which may also include an example SparseMatrix. Now we only have an example for DenseMatrix.

**Pull request closed without merging**